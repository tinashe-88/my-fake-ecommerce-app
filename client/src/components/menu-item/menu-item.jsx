import React from 'react'

import { withRouter } from 'react-router-dom'

import {
  MenuItemContainer,
  BackgroundImage,
  ContentContainer,
  TitleContainer,
  SubtitleContainer
} from './menu-item.styles'

export const MenuItem = ({ 
  title, imageUrl, size, history, linkUrl, match 
}) => (
  <MenuItemContainer
    size={`${size}`}
    onClick={() => history.push(`${match.url}${linkUrl}`)}
  >
    <BackgroundImage 
      imageUrl={imageUrl}
      className='background-image'
    />
    <ContentContainer
      className='content'
    >
      <TitleContainer>{title}</TitleContainer>
      <SubtitleContainer>Shop Now</SubtitleContainer>
    </ContentContainer>
  </MenuItemContainer>
)

export default withRouter(MenuItem)