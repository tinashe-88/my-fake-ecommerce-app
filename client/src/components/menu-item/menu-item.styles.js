import styled from 'styled-components'

export const MenuItemContainer = styled.div`
  height: ${({ size }) => (size ? '380px' : '240px')};
  min-width: 30%;
  overflow: hidden;
  flex: 1 1 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 7.5px 15px;

  &:hover {
    cursor: pointer;

    & .background-image {
      transform: scale(1.1);
      transition: transform 6s cubic-bezier(0.215, 0.610, 0.355, 1)
    }

    & .content {
      opacity: 0.9;
    }
  }
    
  &:first-child {
    margin-right: 7.5px;
  }
    
  &:last-child {
    margin-left: 7.5px;
  }

  @media screen and (max-width: 800px) {
    height: 200px;
  }
`

export const BackgroundImage = styled.div`
  height: 100%;
  width: 100%;
  background-position: center;
  background-size: cover;
  background-image: ${({ imageUrl }) => `url(${imageUrl})`};
`

export const ContentContainer = styled.div`
  height: 90px;
  padding: 0 25px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
  background-color: #fff;
  opacity: 0.8;
  position: absolute;

  @media screen and (max-width: 500px) {
    width: 37%;
  }
`

export const TitleContainer = styled.h1`
  font-weight: bold;
  margin-bottom: 6px;
  font-size: 22px;
  color: #4a4a4a;
`

export const SubtitleContainer = styled.span`
  font-weight: lighter;
  font-size: 16px;
`