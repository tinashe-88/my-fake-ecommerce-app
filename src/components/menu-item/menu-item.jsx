import React from 'react'

import {
  MenuItemContainer,
  BackgroundImage,
  ContentContainer,
  TitleContainer,
  SubtitleContainer
} from './menu-item.styles'

const MenuItem = ({ title, imageUrl, size }) => (
  <MenuItemContainer
    className={`${size}`}
  >
    <BackgroundImage 
      style={{
        backgroundImage: `url(${imageUrl})`
      }}
    />
    <ContentContainer>
      <TitleContainer>{title}</TitleContainer>
      <SubtitleContainer>Shop Now</SubtitleContainer>
    </ContentContainer>
  </MenuItemContainer>
)

export default MenuItem