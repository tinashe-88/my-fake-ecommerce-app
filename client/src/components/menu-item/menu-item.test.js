import React from 'react'
import { shallow } from 'enzyme'

import { MenuItem } from './menu-item'

describe('MenuItem component', () => {
  let wrapper
  let mockMatch
  let mockHistory

  const linkUrl = '/jackets'
  const size = 'large'
  const imageUrl = 'image'

  beforeEach(() => {
    mockMatch = {
      url: '/shop'
    }

    mockHistory = {
      push: jest.fn()
    }

    const mockProps = {
      title: 'jackets',
      imageUrl: imageUrl, 
      size, 
      history: mockHistory, 
      linkUrl, 
      match: mockMatch
    }

    wrapper = shallow(<MenuItem {...mockProps}/>)
  })

  it('Should render MenuItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call history.push with the correct string when MenuItemContainer is clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click')
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`)
  })

  it('Should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size)
  })

  it('Should pass imageUrl to BackgroundImage as the prop imageUrl', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(imageUrl)
  })
})