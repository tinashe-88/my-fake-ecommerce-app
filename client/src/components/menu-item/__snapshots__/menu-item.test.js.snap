// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component Should render MenuItem component 1`] = `
<MenuItemContainer
  onClick={[Function]}
  size="large"
>
  <BackgroundImage
    className="background-image"
    imageUrl="image"
  />
  <styled.div
    className="content"
  >
    <styled.h1>
      jackets
    </styled.h1>
    <styled.span>
      Shop Now
    </styled.span>
  </styled.div>
</MenuItemContainer>
`;
